packages/calypso-e2e/src/lib/pages/stats-page.ts

```diff
@@ -113,7 +113,12 @@ export class StatsPage {
 		// Wait for the charts to load first, even if no activity is present.
 		// CSS selector has to be used here because there is no a11y selector
 		// for this element.
-		await this.page.locator( selectors.graph ).waitFor();
+		// A loop is used here because multiple locators can match the CSS locator
+		// and to be safe, it's best to wait for all placeholders to be detached.
+		const locators = await this.page.locator( '.is-loading' ).all();
+		for ( const locator of locators ) {
```

Comment:
**thought (non-blocking):** While I understand why a loop is used here, I think using `Promise.all()` is more efficient because it waits for all elements to be detached concurrently rather than sequentially. What do you think?

Reply:
Good point, I think I've subconsciously tried to stop using `Promise.all`.

However, I'm drawing a bit of a blank on how to apply the `waitFor()` method to the array produced by the call in Line 118. I don't think I've ever done that with a locator before - with a `page.waitForSelector`, yes. It's easy to do with the `waitForSelector`:

```
await page.waitForSelector( <selector> );
```